### PR TITLE
New bombardment weapons

### PR DESCRIPTION
**Pull request by ghost**
EMP Generator (kills robotic)
Graviton Pulse (kills lithic)
Dark Ray (kills phototropic)
Chaos Wave (kills everything)

### CODE REVIEW

**Comment by Vezzra** [comment]
Hm, looks interesting. @MatGB, can you take a look at this? I've to go over the thing a bit more thoroughly, but at a first glance it looks promising.

File: default/ship_parts.txt [hunk 1]
```diff
@@ -984,6 +984,228 @@ Part
...
+                SetPopulation value = Value - 0.5
+                GenerateSitRepMessage
+                    message = "EFFECT_BIOWEAPON"
+                    parameters = [
+                        tag = "system" data = Source.System.ID
+                    ]
+                    empire = Target.Owner
+            ]
+
+        EffectsGroup    // monsters use death spores on player planets
```
Comment (MatGB) [inline default/ship_parts.txt]: I think this effectsgroup can be removed, if we're sticking with spores and terminators as generics we don't need a monster use clause as the new ones aren't used by monsters?

But the comment definitely needs changing as this part isn't death spores.

File: default/ship_parts.txt [hunk 2]
```diff
@@ -984,6 +984,228 @@ Part
...
+                OrderedBombardedBy condition = Source
+                Or [
+                    OwnedBy affiliation = EnemyOf empire = Source.Owner
+                    Unowned
+                ]
+                VisibleToEmpire empire = Source.Owner
+            ]
+            effects = [
+                SetPopulation value = Value - 2
+                RemoveSpecial name = "GAIA_SPECIAL"
```
Comment (MatGB) [inline default/ship_parts.txt:1261]: Ooh, I like this bit, not sure on the balance, but good call.

File: default/stringtables/en.txt [hunk 3]
```diff
@@ -8679,14 +8685,38 @@ SHP_DEATH_SPORE
 Death Spores
 
 SHP_DEATH_SPORE_DESC
-Unlocks the [[shippart SP_DEATH_SPORE]] which allows ships to reduce the population of enemy planets.
+Unlocks the [[shippart SP_DEATH_SPORE]] which allows ships to reduce the Organic population of enemy planets.
```
Comment (MatGB) [inline default/stringtables/en.txt:8748]: Ah. You haven't reverted the death spores/bio terminators back to organic only hence my partial confusion earlier.

**Comment by MatGB** [comment]
Overall, I like the idea and this implementation with the branching tech tree with different prerequisites is a nice approach.

I have balance concerns, not least that in trunk death spores and bio terminators are for all species and that isn't changed in this, then there's the additional problem that this, if it is changed, would give two different organic weapons and one for each of the others.

It's good work, but the comment text needs adjusting, we need to decide how to deal with monsters (I like that Kraken and Great Kraken can affect all species types).

Perhaps death spores affect all, bio terminators just organics (and possibly swap them name/strength wise?) Or rename bio terminators so they do 1 damage each and the new chaos thing does 2?

Ooh, perhaps make the wave a Core slot instead?

**Comment by MatGB** [comment]
To clarify: can't merge as is, typos and copy/paste errors need fixing at a minimum, I'm worried about knock on effects but not massively, and it's good work overall.

File: default/ship_parts.txt (hunk 1 above)
Comment (ghost) [inline default/ship_parts.txt]: Then intent was that these could be used by monsters in future.
Comment fixed.  Good catch.

File: default/stringtables/en.txt (hunk 3 above)
Comment (ghost) [inline default/stringtables/en.txt:8748]: Good catch.  I hadn't realized that had been changed.  Thanks.

**Comment by ghost** [comment]
I'm trying to come up with some additional weapons to round it out, as MatGB suggested, then it can be balanced.  (RP cost and damage.)  Assignment to monsters should only be done once the weapons themselves are finalized.
I am against making Chaos Wave a core slot, as I think that would be too limiting.  Maybe it should be more expensive?  It should also be restricted to Experimenters, or other late-game monsters, as it is too destructive for early game.

**Comment by ghost** [comment]
First tier weapons all do 0.5 damage, and are the same depth in the tech tree.
Second tier all do 1 damage, and are in the next two tech depths.
Chaos Wave does 2 damage and depends on all other bombard techs, but it should probably depend on another.  Maybe Death Ray?
I have EMP Generator and Sonic Shockwave both depending on Force-Field Harmonics, which I'm not too happy with, but I don't see better parent techs for either one.

**Comment by Vezzra** [comment]
Ok, got a PM from @spikethehobbitmage that this patch is ready now, will try to do some review and testing next week. @MatGB, can you give this PR another pass too? Four eyes might catch more than two... :wink:

**Comment by MatGB** [comment]
Heh, had this open to look up, my Git GUI client has crashed on me twice while I try to grab this branch but I will do ASAP, busy this week tho so no immediate promises.

File: default/techs.txt [hunk 4]
```diff
@@ -2751,6 +2753,113 @@ Tech
...
+    name = "SHP_EMO"
+    description = "SHP_EMO_DESC"
+    short_description = "SHIP_PART_UNLOCK_SHORT_DESC"
+    techtype = Application
+    category = "SHIPS_CATEGORY"
+    researchcost = 75 * [[TECH_COST_MULTIPLIER]]
+    researchturns = 5
+    prerequisites = [
+	"SHP_EMP"
+	"SHP_FRC_ENRG_COMP"
```
Comment (Vezzra) [inline default/techs.txt]: Code formatting nitpick: the above two lines are indented by one tab, pls replace with 8 whitespaces instead.

File: default/techs.txt [hunk 5]
```diff
@@ -2751,6 +2753,113 @@ Tech
...
+    name = "SHP_SONIC"
+    description = "SHP_SONIC_DESC"
+    short_description = "SHIP_PART_UNLOCK_SHORT_DESC"
+    techtype = Application
+    category = "SHIPS_CATEGORY"
+    researchcost = 450 * [[TECH_COST_MULTIPLIER]]
+    researchturns = 3
+    prerequisites = [
+        "SHP_BOMBARD"
+	"LRN_FORCE_FIELD"
```
Comment (Vezzra) [inline default/techs.txt]: Same for above line: replace tab with 8 spaces.

File: default/techs.txt [hunk 6]
```diff
@@ -2751,6 +2753,113 @@ Tech
...
+    name = "SHP_VOID_SHADOW"
+    description = "SHP_VOID_SHADOW_DESC"
+    short_description = "SHIP_PART_UNLOCK_SHORT_DESC"
+    techtype = Application
+    category = "SHIPS_CATEGORY"
+    researchcost = 75 * [[TECH_COST_MULTIPLIER]]
+    researchturns = 5
+    prerequisites = [
+        "SHP_DARK_RAY"
+	"LRN_GATEWAY_VOID"
```
Comment (Vezzra) [inline default/techs.txt]: And again for above line: replace tab with 8 spaces.

File: default/techs.txt [hunk 7]
```diff
@@ -2751,6 +2753,113 @@ Tech
...
+    ]
+    unlock = Item type = ShipPart name = "SP_DARK_RAY"
```
Comment (Vezzra) [inline default/techs.txt]: Shouldn't that be SP_VOID_SHADOW here...?

**Comment by Vezzra** [comment]
Ok, went over the patch & found a few things, see my line notes above. Other than that, looks fine as far as I can tell. My quick test ran fine too, so after you fix the remaining issues and Mat gives his ok, this can me merged.

File: default/techs.txt (hunk 7 above)
Comment (ghost) [inline default/techs.txt]: Correct.  Fixed.

**Comment by Vezzra** [comment]
> Heh, had this open to look up, my Git GUI client has crashed on me twice while I try to grab this branch but I will do ASAP, busy this week tho so no immediate promises.

@MatGB, do you think you will be able to get around to check this PR anytime soon, or shall I just merge it?

**Comment by MatGB** [comment]
To be honest I've been busy and distracted by the fighters, if you think it's fine go ahead it's basically a sound idea, and it might add some impetus to my overall attempts to balance weaponry a bit more.

File: default/space_monsters.txt [hunk 8]
```diff
@@ -397,7 +397,7 @@ ShipDesign
     hull = "SH_BLOATED_JUGGERNAUT_BODY"
     parts = [
         ""
-        "SP_BIOTERM"
```
Comment (Cjkjvfnby) [inline default/space_monsters.txt:400]: Sorry cannot che k it from mobile, is old key in use?

Comment (Vezzra) [inline default/space_monsters.txt:400]: I'm not exactly sure what you're referring to... do you mean if "SP_BIOTERM" is still in use? If that's what you want to know, then yes. Why?

**Pull request merged**